### ISSUE TITLE
Using Gstreamer authentication issue

### ISSUE THREAD

**Issue by CACUser** [aler9/rtsp-simple-server#158]
I am trying to follow the Readme to publish using Gstreamer. I tried two ways of using docker:

docker run --rm -it-network=host54 aler9/rtsp-simple-server 

 docker run --rm -it -e RTSP_PROTOCOLS=tcp -p 8554:8554 aler9/rtsp-simple-server 

I am getting the following when I publish:
gst-launch-1.0 rtspclientsink name=s location=rtsp://localhost:8554/mystream filesrc 
location=test.mp4 ! qtdemux name=d d.video_0 ! queue ! s.sink_0 d.audio_0 ! queue ! s.sink_1
Setting pipeline to PAUSED ...
Pipeline is PREROLLED ...
Progress: (open) Opening Stream
Progress: (connect) Connecting to rtsp://localhost:8554/mystream
WARNING: from element /GstPipeline:pipeline0/GstQTDemux:d: Delayed linking failed.
Additional debug info:
./grammar.y(510): gst_parse_no_more_pads (): /GstPipeline:pipeline0/GstQTDemux:d:
failed delayed linking pad  audio_0 of GstQTDemux named d to some pad of GstQueue named queue1
Progress: (open) Retrieving server options
Progress: (open) Opened Stream
Setting pipeline to PLAYING ...
New clock: GstSystemClock
Progress: (request) Sending RECORD request

And if I try reading I get authentication error:
gst-launch-1.0 rtspsrc location=rtsp://localhost:8554/mystream name=s
 s. ! application/x-rtp,media=video ! decodebin ! autovideosink s. ! application/x-rtp,media=a
udio ! decodebin ! audioconvert ! audioresample ! autoaudiosink
Setting pipeline to PAUSED ...
XDG_RUNTIME_DIR (/run/user/1000) is not owned by us (uid 0), but by uid 1000! (This could e g happen if you try to connect to a non-root PulseAudio as a root user, over the native protocol. 
Don't do that.)
XDG_RUNTIME_DIR (/run/user/1000) is not owned by us (uid 0), but by uid 1000! (This could e g happen if you try to connect to a non-root PulseAudio as a root user, over the native protocol. 
Don't do that.)
pci id for fd 9: 1a03:2000, driver (null)
Pipeline is live and does not need PREROLL ...
WARNING: from element /GstPipeline:pipeline0/GstAutoAudioSink:autoaudiosink0: Failed to connect: Connection refused
Additional debug info:
pulsesink.c(615): gst_pulseringbuffer_open_device (): /GstPulseSink:autoaudiosink0-actual-sink-pulse
Got context from element 'autovideosink0': gst.gl.GLDisplay=context, gst.gl.GLDisplay=(GstGLDisplay)"\(GstGLDisplayGBM\)\ gldisplaygbm0";
Progress: (open) Opening Stream
Progress: (connect) Connecting to rtsp://localhost:8554/mystream
Progress: (open) Retrieving server options
Progress: (open) Retrieving media info
ERROR: from element /GstPipeline:pipeline0/GstRTSPSrc:s: Could not open resource for reading.
Additional debug info:
gstrtspsrc.c(5829): gst_rtspsrc_setup_auth (): /GstPipeline:pipeline0/GstRTSPSrc:s:
No supported authentication protocol was found
ERROR: pipeline doesn't want to preroll.
Setting pipeline to PAUSED ...
Setting pipeline to READY ...
Setting pipeline to NULL ...
Freeing pipeline ...

Am I missing anything? I did follow the instructions in the README. Do I need to change any config options?

Thanks!

**Comment by dorinclisu** [comment]
> WARNING: from element /GstPipeline:pipeline0/GstQTDemux:d: Delayed linking failed.
Your first pipeline for publishing seems to be failing, and this has nothing to do with rtsp server. Try with gst-launch-1.0 -v to possibly get more clues or set GST_DEBUG environment variable to "DEBUG".

Since you are not publishing anything, of course the second pipeline also fails to read, and again this has little to do with the rtsp server itself.

**Comment by aler9** [comment]
In my opinion, the problem is that your sample file doesn't contain an audio track, but only a video track, therefore the pipeline must be modified to transmit the video track only

```
gst-launch-1.0 rtspclientsink name=s location=rtsp://localhost:8554/mystream filesrc
location=test.mp4 ! qtdemux name=d d.video_0 ! queue ! s.sink_0
```

**Comment by CACUser** [comment]
I have tried with above pipeline and am seeing the same issue. So I changed test file to a video with audio. and turend on debug level to 2 for Gstreamer. The ffmpeg pipeline work beautifully. Maybe there is an issue with my Gstreamer setup.
but here's the log I get with a video with audio:
gst-launch-1.0 rtspclientsink name=s location=rtsp://localhost:8554/mystream filesrc location=bigbuckbunny.mp4 ! qtdemux name=d d.video_0 ! queue ! s.sink_0 d.audio_0 ! queue ! s.sink_1
Setting pipeline to PAUSED ...
0:00:00.020727437  7179 0x55f6788fed50 WARN                 basesrc gstbasesrc.c:3583:gst_base_src_start_complete:<filesrc0> pad not activated yet
Pipeline is PREROLLED ...
Progress: (open) Opening Stream
0:00:00.021315859  7179 0x55f6789018a0 WARN                 qtdemux qtdemux.c:3031:qtdemux_parse_trex:<d> failed to find fragment defaults for stream 1
0:00:00.021455998  7179 0x55f6789018a0 WARN                 qtdemux qtdemux.c:3031:qtdemux_parse_trex:<d> failed to find fragment defaults for stream 2
Progress: (connect) Connecting to rtsp://localhost:8554/mystream
Progress: (open) Retrieving server options   
Progress: (open) Opened Stream
Setting pipeline to PLAYING ...
New clock: GstSystemClock
Progress: (request) Sending RECORD request   
Progress: (record) Sending server stream info
Progress: (request) SETUP stream 1
0:00:00.043328908  7179 0x55f678901a80 ERROR         rtspclientsink gstrtspclientsink.c:3679:gst_rtsp_client_sink_create_transports_string:<s> No supported profiles configured
0:00:00.043367911  7179 0x55f678901a80 WARN          rtspclientsink gstrtspclientsink.c:4119:gst_rtsp_client_sink_setup_streams:<s> error: Could 
not setup transport.
0:00:00.043422639  7179 0x55f678901a80 ERROR         rtspclientsink gstrtspclientsink.c:4430:gst_rtsp_client_sink_record:<s> setup failed        
ERROR: from element /GstPipeline:pipeline0/GstRTSPClientSink:s: Could not get/set settings from/on resource.
Additional debug info:
gstrtspclientsink.c(4119): gst_rtsp_client_sink_setup_streams (): /GstPipeline:pipeline0/GstRTSPClientSink:s:
Could not setup transport.
Execution ended after 0:00:00.009406937
Setting pipeline to PAUSED ...
Setting pipeline to READY ...
0:00:00.044773475  7179 0x55f678901a80 WARN          rtspclientsink gstrtspclientsink.c:2893:gst_rtsp_client_sink_try_send:<s> receive interrupted
0:00:00.044844801  7179 0x55f678901a80 WARN          rtspclientsink gstrtspclientsink.c:3272:gst_rtsp_client_sink_open:<s> Failed to connect to server
Setting pipeline to NULL ...
Freeing pipeline ...
